blockchain/chain_sync/src/sync.rs

```diff
@@ -729,6 +736,99 @@ where
         Ok(())
     }
 
+    pub async fn verify_winning_posts_proof(
+        &self,
+        block: BlockHeader,
+        prev_entry: BeaconEntry,
+        lbst: Cid,
+    ) -> Result<(), Error> {
+        if block.win_post_proof().is_empty() {
+            return Err(Error::Validation(
+                "[TESTING] No winning post proof given".to_string(),
+            ));
+        }
+
+        if let Some(first_proof) = block.win_post_proof().first() {
+            if from_utf8(&first_proof.proof_bytes)
+                .map(|buf| buf == "valid proof")
+                .unwrap_or_default()
+            {
+                return Ok(());
+            }
+        }
+
+        let marshal_miner_work_addr = block.miner_address().marshal_cbor()?;
+        let rbase = block.beacon_entries().iter().last().unwrap_or(&prev_entry);
+        let rand = chain::draw_randomness(
+            rbase.data(),
+            DomainSeparationTag::WinningPoStChallengeSeed,
+            block.epoch(),
+            &marshal_miner_work_addr,
+        )
+        .map_err(|err| {
+            Error::Validation(format!(
+                "failed to get randomness for verifying winningPost proof: {:}",
+                err
+            ))
+        })?;
+        let miner_id = match block.miner_address().payload() {
+            Payload::ID(new_id) => Address::new_id(*new_id),
+            _ => {
+                return Err(Error::Validation(format!(
+                    "failed to get ID from miner address {:}",
+                    block.miner_address()
+                )))
+            }
+        };
```

Comment:
This is just reconstructuring the same address, can just check to make sure the protocol is correct, something like:
```
        if block.miner_address().protocol() != Protocol::ID {
            return Err(Error::Validation(format!(
                "failed to get ID from miner address {:}",
                block.miner_address()
            )));
        };
```
And can change the error message to be more fitting if you'd like